flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/runtime/functions/SqlFunctionUtils.java

```diff
@@ -1174,6 +1174,148 @@ public static String toBase64(byte[] bytes) {
         return Base64.getEncoder().encodeToString(bytes);
     }
 
+    public static BinaryStringData jsonQuote(BinaryStringData input) {
+        if (input == null) {
+            return null;
+        }
+
+        StringBuilder result = new StringBuilder("\"");
+
+        String inputStr = input.toString();
+        for (int i = 0; i < inputStr.length(); ) {
+            int codePoint = inputStr.codePointAt(i);
+            int charCount = Character.charCount(codePoint);
+
+            switch (codePoint) {
```

Comment:
Escape doesn't seem to follow JSON spec defined at [1]
e.g. it says that `*`, `/` should be escaped and here it doesn't escape it

[1] https://www.json.org/json-en.html

Reply:
Here I also used MySQL reference [1]. I am ok to use json.org as a reference

[1] https://dev.mysql.com/doc/refman/8.0/en/json-modification-functions.html#json-unquote-character-escape-sequences